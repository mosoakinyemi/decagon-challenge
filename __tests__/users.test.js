import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import Users from '../src/screens/users';
global.fetch = jest.fn(async () =>
  Promise.resolve({
    json: () => Promise.resolve({rates: {CAD: 1.42}}),
  }),
);

jest.useFakeTimers();

describe('<Users />', () => {
  it('renders correctly', () => {
    const instance = renderer.create(<Users />);
    expect(instance).toMatchSnapshot();
    spyOn(console, 'error');
  });
});
