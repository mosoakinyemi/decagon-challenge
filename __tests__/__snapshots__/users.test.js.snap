// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d8d8e8",
        "elevation": 5,
        "flexDirection": "row",
        "height": 90.5,
        "justifyContent": "space-between",
        "paddingHorizontal": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "opacity": 1,
          "width": 110,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 26.5,
            },
            undefined,
            Object {
              "fontFamily": "Entypo",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï…‰
      </Text>
      <Text
        style={
          Object {
            "color": "#2f3138",
            "fontSize": 29.5,
            "lineHeight": 44.5,
            "marginLeft": 10,
          }
        }
      >
        Back
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#2f3138",
          "fontSize": 29.5,
          "fontWeight": "bold",
          "lineHeight": 44.5,
        }
      }
    >
      Contacts
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "opacity": 1,
          "width": 110,
        }
      }
    />
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgba(255,255,255,0.7)",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#2f3138"
      hidesWhenStopped={true}
      size="large"
    />
    <Text
      style={
        Object {
          "color": "#2f3138",
          "fontSize": 29.5,
          "fontWeight": "bold",
          "lineHeight": 44.5,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
